### PR TITLE
busybox: don't install NTP scripts if NTP isn't configured

### PR DESCRIPTION
**Pull request by pprindeville**
If you're using Chrony or NTPD you don't want the busybox NTP server
as well.  Make it's installation truly conditional.

Signed-off-by: Philip Prindeville <philipp@redfish-solutions.com>

### CODE REVIEW

**Comment by pprindeville** [comment]
@nbd168 @hnyman @stintel @jow- @hauke @dedeckeh @sbyx @blogic

File: package/utils/busybox/Makefile [hunk 1]
```diff
@@ -110,12 +110,19 @@ define Build/Compile
...
+ifneq ($(CONFIG_BUSYBOX_CONFIG_CROND),)
 	$(INSTALL_BIN) ./files/cron $(1)/etc/init.d/cron
+endif
+ifneq ($(CONFIG_BUSYBOX_CONFIG_NTPD),)
 	$(INSTALL_BIN) ./files/sysntpd $(1)/etc/init.d/sysntpd
 	$(INSTALL_BIN) ./files/ntpd-hotplug $(1)/usr/sbin/ntpd-hotplug
-	-rm -rf $(1)/lib64
+endif
+	rm -rf $(1)/lib64
 endef
```
Comment (blogic) [inline package/utils/busybox/Makefile]: removing the dash is an unrelated change

Comment (pprindeville) [inline package/utils/busybox/Makefile]: Fixed.

**Comment by nbd168** [comment]
Pushed a simplified version of this to my staging tree, thanks.

**Comment by pprindeville** [comment]
Thanks. Can you molease attach a link to your commit so I can see the final version? Thanks

**Comment by dedeckeh** [comment]
@pprindeville https://git.lede-project.org/?p=lede/nbd/staging.git;a=commit;h=5609b039cfced708a3acc3775824a2da10bbfe1c

**Comment by pprindeville** [comment]
@nbd168: hang on, looking over this again:

```
+ifneq ($(CONFIG_BUSYBOX_CONFIG_NTPD),)
        $(INSTALL_BIN) ./files/sysntpd $(1)/etc/init.d/sysntpd
        $(INSTALL_BIN) ./files/ntpd-hotplug $(1)/usr/sbin/ntpd-hotplug
+endif
```

I realize that both `/etc/init.d/sysntpd` (busybox) and `/etc/init.d/ntpd` (NTPd) use helpers (`/usr/sbin/ntpd.hotplug-helper` and `/usr/sbin/ntpd.hotplug-helper`)... plus NTPd uses something else additionally called `/etc/hotplug.d/iface/20-ntpd`.

Is there any redundancy we can fix here?

**Comment by arjendekorte** [comment]
It looks like the logic is reversed here. The scripts are now installed when CONFIG_BUSYBOX_CONFIG_CROND and CONFIG_BUSYBOX_CONFIG_NTPD are not configured.

**Comment by nbd168** [comment]
The logic wasn't reversed, but it wasn't checking for the correct symbol names in some cases. I've pushed a fix for that.
